wallet/src/de/schildbach/wallet/WalletApplication.java

```diff
@@ -324,6 +330,18 @@ public void setWallet(Wallet newWallet) {
         if (!wallet.hasKeyChain(Constants.BIP44_PATH)) {
             wallet.addKeyChain(Constants.BIP44_PATH);
         }
+        if (!authenticationGroupExtension.hasKeyChains()) {
+            authenticationGroupExtension.addKeyChains(
+                    wallet.getParams(),
+                    wallet.getKeyChainSeed(),
+                    EnumSet.of(
+                            AuthenticationKeyChain.KeyChainType.MASTERNODE_OWNER,
+                            AuthenticationKeyChain.KeyChainType.MASTERNODE_VOTING,
+                            AuthenticationKeyChain.KeyChainType.MASTERNODE_OPERATOR,
+                            AuthenticationKeyChain.KeyChainType.MASTERNODE_PLATFORM_OPERATOR
+                    )
+            );
+        }
```

Comment:
We should separate wallet creation into its own class.  We have many classes that create wallet objects based on what the user is doing.  And there are many static helper methods.  Currently there is wallet creation code sprinkled about in many places and probably inconsistently.